### PR TITLE
YOLOv9 학습 파라메터 조정 및 버그 수정

### PR DESCRIPTION
**Pull request by HyunwooCho**
### PR 타입
하나 이상의 PR 타입을 선택해주세요.
- [ ] 기능 추가
- [ ] 기능 삭제
- [X] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트
- [ ] 기타

### PR전 체크 사항
PR전 아래 사항을 확인하였는지 체크하시고 PR 생성 바랍니다.
- [X] 자신의 브랜치로 main 브랜치 최신 내용 반영후 동작 테스트 완료
- [X] 다른 작업자의 폴더내 파일 수정하지 않음.
- [X] docker-compose.yml 수정한 경우, 수정에 따른 영향 평가 

### 관련 이슈
#215 

### PR 반영 브랜치
ETRI-tenace-AutoNN -> main

### 변경 사항
- YOLOv9 학습을 위한 hyper-parameters 및 arguments의 default value 변경  
    1. 최종 학습률 lrf: 0.1 -> 0.01
    2. 증강 학습 translate: 0.2 -> 0.1
    3. 모자이크 증강 학습 멈추는 시점 close_mosaic 추가: 15
    4. 학습률 변화 곡선: linear_lr : False -> True

- Loss / Metric 계산 관련 수정 사항

    1. Class, DFL, Box gain factor 값 조정 (각각 gain factor를 곱하여 합산한 값이 최종 loss)
    2. xywh-to-xyxy 좌표 변환 인덱싱 오류 수정

- Optimizer 관련 수정 사항

    1. group mapping 오류 수정
    2. warm-up epoch 내에서도 optimizer grouping 수정
    3. Optimizer scheduler: One-cycle LR -> Linear LR로 수정
    4. Linear LR 수식 오류 수정: epoch-1로 나누는 수식 -> epoch로 나누도록 수정(epoch = 1 설정시 오류 방지)

- warm-up epoch 최소값 수정: 1000 -> 100
- dataset loading 관련 수정: close mosaic 도입으로 infinite data loader -> normal data loader
- model initialization 관련 수정: nn.SiLU inplace = True 설정

- 시험 재현성을 위한 random seed 설정 관련

    1. random seed 기본 값 1 -> 0
    2. pytorch Generator() 함수 활용 -> 데이터셋 구성에 입력 인자로 주입
    3. seed_worker() 구현
    4. deterministic 설정 가능한 random generator 구현
 
### 테스트 결과
YOLOv9-T를 모든 하이퍼파라메터와 옵션을 동일하게 주고 scratch부터 학습하였을 때, 
original YOLOv9-T와 완전히 동일한 시험 결과를 보임을 확인함

### CODE REVIEW

**Review by ML-TANGO — APPROVED** [review]
PR 내용 및 관련 이슈 확인후 승인
